nixos/modules/services/logging/filebeat.nix

```diff
@@ -229,10 +229,11 @@ in
 
           umask u=rwx,g=,o=
 
-          ${utils.genJqSecretsReplacementSnippet
-              cfg.settings
-              "/var/lib/filebeat/filebeat.yml"
-           }
+          ${utils.genConfigOutOfBand {
+              config = cfg.settings;
+              configLocation = "/var/lib/filebeat/filebeat.yml";
+              generator = utils.genConfigOutOfBandFormatAdapter json;
```

Comment:
Shouldn't it be `yaml`? I assume the module used `json` due to `genJqSecretsReplacementSnippet`.

Reply:
You know what, I tried YAML at first but it failed. It's because of the private key, it includes newlines. So with YAML, the newlines was creating a syntax error because the indentation was not respected. With JSON, the syntax is valid even with newlines because it includes opening and closing quotes. And anyway, JSON is a subset of YAML afterall.

That being said, another solution would be to enhance the `replaceSecret.py` script to have it replace "real" newlines with `\n`. That would be fully backwards compatible. I didn't do it because I don't want the replaceSecret.py script to have one argument per use case but could if you'd think that's better.